code/modules/admin/centcomviewer.dm

```diff
@@ -0,0 +1,46 @@
+// API Documentation - https://centcom.melonmesa.com/swagger/index.html
+/// Admin CentCom Viewer Panel for looking up public bans on other servers via the CentCom Viewer API
+/datum/centcomviewer
+	/// whether or not we are filtering inactive bans from view
+	var/filterInactive = FALSE
+	///target key of the user, the centcom api will make it a ckey on its own
+	var/target_key
+
+/datum/centcomviewer/ui_state(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/centcomviewer/ui_status(mob/user)
+  return tgui_admin_state.can_use_topic(src, user)
+
+/datum/centcomviewer/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "CentComViewer")
+		ui.open()
+	else
+		update_static_data(user, ui)
```

Comment:
what's this for? this shouldn't be needed

Reply:
if you already have the panel and then try to open another for a different player it wont fresh unless you hit f5. This forces it to grab the new ban data and target key